### ISSUE TITLE
Role grants with multiple users and roles not applying correctly

### ISSUE THREAD

**Issue by DVAlexHiggs** [Snowflake-Labs/terraform-provider-snowflake#1629]
**Provider Version**

0.58.2

**Terraform Version**

0.4.0

**Describe the bug**

When creating a `snowflake_role_grants` resource where there are multiple users and multiple roles, `terraform apply` will try to modify the resource every time it is run, even if it has just applied changes in a previous `terraform apply`.

**Expected behavior**

Terraform will create the grant for roles and users on `terraform apply`, and when run again will not attempt to make further changes.

**Code samples and commands**

Given the following resource definition:

```hcl
resource "snowflake_role_grants" "developer_grants" {
  role_name = snowflake_role.developer.name

  roles = [
    "BOB_GELDOF",
    "JERRY_SMITH"
  ]

  users = [
    "BOB_GELDOF",
    "JERRY_SMITH"
  ]
}
```

On the first execution of `terraform apply` terraform will try to do the following:

id: `DEVELOPER|BOB_GELDOF|BOB_GELDOF`
- `roles: + "JERRY_SMITH" # (1 unchanged element hidden)`
- `users: + "JERRY_SMITH" # (1 unchanged element hidden)`

Having applied the above, on the second execution of `terraform apply` terraform will try to do the following:

id: `DEVELOPER|BOB_GELDOF|BOB_GELDOF`
- `roles: + "JERRY_SMITH" # (1 unchanged element hidden)`
- `users: + "JERRY_SMITH" # (1 unchanged element hidden)`

**Additional context**

I believe this is happening because the resource id does not include all users and all roles in the list.

i.e. instead of `DEVELOPER|BOB_GELDOF|BOB_GELDOF` it should be `DEVELOPER|BOB_GELDOF,JERRY_SMITH|BOB_GELDOF,JERRY_SMITH`

This is made more obvious when I try to do the same thing on a resource using `for_each` with single item lists, instead of just lists. This is because the id changes each time to the single item. 

Oddly, in the docs, the **Import** section does have comma delimited roles and users in the id.

**Comment by KPGR** [comment]
Having the same problem. Started after upgrading to 0.58.2.

**Comment by sfc-gh-swinkler** [comment]
Hello, this has been fixed in 0.59

**Comment by DVAlexHiggs** [comment]
Hi @sfc-gh-swinkler Can you reopen this please? This has not been fixed in 0.59.0 and It's a different case
